msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/stress/AbstractMsalUiStressTest.java

```diff
@@ -0,0 +1,267 @@
+import com.microsoft.identity.client.ui.automation.performance.PerformanceProfile;
+import com.microsoft.identity.client.ui.automation.rules.RulesHelper;
+import com.microsoft.identity.client.ui.automation.utils.CommonUtils;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+import com.microsoft.identity.internal.testutils.labutils.LabUserQuery;
+
+import org.junit.Before;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.RejectedExecutionHandler;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+
+public abstract class AbstractMsalUiStressTest<T, S> extends AbstractMsalUiTest {
+
+    // Sets the interval duration in seconds to which the device performance will be monitored.
+    private static final long DEVICE_MONITOR_INTERVAL = TimeUnit.SECONDS.toMillis(1);
+    private static final String DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+    private Exception executionException;
+
+    private final AtomicInteger testsPassed = new AtomicInteger(0);
+    private final AtomicInteger testsFailed = new AtomicInteger(0);
+
+    @Override
+    @Before
+    public void setup() {
+        super.setup();
+        DeviceMonitor.setProfiler(PerformanceProfile.CPU, new CPUMonitor());
+        DeviceMonitor.setProfiler(PerformanceProfile.MEMORY, new MemoryMonitor());
+        DeviceMonitor.setProfiler(PerformanceProfile.NETWORK, new NetworkUsageMonitor());
+    }
+
+    /**
+     * Run the stress tests
+     */
+    public void run() throws Exception {
+        final T prerequisites = this.prepare();
+
+        final BlockingQueue<Runnable> blockingQueue = new ArrayBlockingQueue<>(getNumberOfThreads());
+        final RejectedExecutionHandler rejectedExecutionHandler = new ThreadPoolExecutor.CallerRunsPolicy();
+
+        final ExecutorService executorService = new ThreadPoolExecutor(
+                1,
+                getNumberOfThreads() + 1,
+                0L,
+                TimeUnit.MILLISECONDS,
+                blockingQueue,
+                rejectedExecutionHandler
+        );
+
+
+        final long startTime = System.currentTimeMillis();
+        final long timeLimit = TimeUnit.MINUTES.toMillis(getTimeLimit());
+
+        // use the execution thread pool to collect device stats
+        executorService.submit(getPerformanceStatsCollector());
+
+
+        for (int i = 0; i < getNumberOfThreads(); i++) {
+            executorService.submit(new Runnable() {
+                @Override
+                public void run() {
+                    while (System.currentTimeMillis() - startTime < timeLimit && executionException == null) {
```

Comment:
I predict we're going to want to continue in the face of exceptions at some point, but this is OK for now.